docs/deployment_patterns/how_to_use_great_expectations_in_flyte.md

```diff
@@ -0,0 +1,202 @@
+def total_pay(hourly_pay: float, hours_worked: int, df: pd.DataFrame) -> pd.DataFrame:
+   return df.assign(total_pay=hourly_pay * hours_worked)
+```
+
+## The Idea
+Flyte helps automate how we architect our ML or data pipelines, which means data is an integral part of Flyte, and lots of data points flow in and out of the pipelines. Keeping an eye on data all the time isn’t a feasible solution. What we need is an automated mechanism that validates our data thoroughly.
+
+Flyte does have a [native type system](https://docs.flyte.org/projects/cookbook/en/latest/auto/core/type_system/flyte_python_types.html) that enforces the correctness of data; however, that alone doesn’t suffice in performing comprehensive data validation. This is where Great Expectations can help Flyte take its data validation system to the next level!
+
+*If you haven’t worked with Flyte before, jump right into the [getting started tutorial](https://docs.flyte.org/en/latest/getting_started.html).*
+
+## Background
+As mentioned earlier, Flyte has a native type system. This includes support for FlyteFile and FlyteSchema. 
+* [FlyteFile](https://docs.flyte.org/projects/flytekit/en/latest/generated/flytekit.types.file.FlyteFile.html#flytekit.types.file.FlyteFile): FlyteFile represents an automatic persistence object in Flyte. It represents files in remote storage, and Flyte will transparently materialize them in every task execution.
+* [FlyteSchema](https://docs.flyte.org/projects/flytekit/en/latest/generated/flytekit.types.schema.FlyteSchema.html#flytekit.types.schema.FlyteSchema): FlyteSchema supports tabular data. Examples include pandas DataFrame, Spark DataFrame, etc. 
+
+The integration has been written such that both FlyteFile and FlyteSchema are inherently supported. So you can validate FlyteFile and FlyteSchema using Great Expectations within any Flyte pipeline!
+
+*Flyte types have been added because, in Great Expectations, we have the privilege to give a non-string (Pandas/Spark DataFrame) when using a [RuntimeDataConnector](https://docs.greatexpectations.io/docs/reference/datasources#runtimedataconnector-and-runtimebatchrequest) but not when using an [InferredAssetFilesystemDataConnector](https://legacy.docs.greatexpectations.io/en/latest/autoapi/great_expectations/datasource/data_connector/inferred_asset_filesystem_data_connector/index.html?highlight=InferredAssetFilesystemDataConnector#great_expectations.datasource.data_connector.inferred_asset_filesystem_data_connector.InferredAssetFilesystemDataConnector) or a [ConfiguredAssetFilesystemDataConnector](https://legacy.docs.greatexpectations.io/en/latest/autoapi/great_expectations/datasource/data_connector/configured_asset_filesystem_data_connector/index.html#great_expectations.datasource.data_connector.configured_asset_filesystem_data_connector.ConfiguredAssetFilesystemDataConnector). For the latter case, with the integration of Flyte types, we can give a Pandas/Spark DataFrame or a remote URI as the dataset.*
```

Comment:
```suggestion
*Flyte types have been added because, in Great Expectations, we have the ability to pass a non-string (Pandas/Spark DataFrame) when using a [RuntimeDataConnector](https://docs.greatexpectations.io/docs/reference/datasources#runtimedataconnector-and-runtimebatchrequest) but not when using an [InferredAssetFilesystemDataConnector](https://legacy.docs.greatexpectations.io/en/latest/autoapi/great_expectations/datasource/data_connector/inferred_asset_filesystem_data_connector/index.html?highlight=InferredAssetFilesystemDataConnector#great_expectations.datasource.data_connector.inferred_asset_filesystem_data_connector.InferredAssetFilesystemDataConnector) or a [ConfiguredAssetFilesystemDataConnector](https://legacy.docs.greatexpectations.io/en/latest/autoapi/great_expectations/datasource/data_connector/configured_asset_filesystem_data_connector/index.html#great_expectations.datasource.data_connector.configured_asset_filesystem_data_connector.ConfiguredAssetFilesystemDataConnector). For the latter case, with the integration of Flyte types, we can give a Pandas/Spark DataFrame or a remote URI as the dataset.*
```